MetaMask/metamask-extension
# feat(mv3): hardware wallet support for trezor

## **Description**

Uses the Offscreen document to add an iframe for Trezor hardware wallet communication. The entry point of the trezor-iframe is trezor-iframe.ts and it is protected by Lavamoat using the build system in the same way that the offscreen document is. 

## **Related issues**

Fixes: #21623 

## **Manual testing steps**

1. run `yarn start:mv3`
2. load the extension into chrome, or reload it if already added.
3. If you previously had your Trezor added to the extension, remove it (or you can add a secondary account from it)
4. Click Add new hardware wallet from the add account dropdown menu. 
5. See that Trezor and QR are valid options
6. Make sure you have Trezor Suite open and your device unlocked. 
7. Click trezor and then connect.
8. See that you are prompted to connect your trezor and select an account to import. 
9. Perform transactions, signatures, etc using the test dapp. 


## **Screenshots/Recordings**



### **Before**
<img width="1225" alt="Screenshot 2023-11-30 at 12 04 49 PM" src="https://github.com/MetaMask/metamask-extension/assets/4448075/e0300d39-561c-4a20-a494-35020c1aa6a9">



### **After**
<img width="778" alt="Screenshot 2023-11-30 at 11 59 01 AM" src="https://github.com/MetaMask/metamask-extension/assets/4448075/bbd1a50c-963e-4eed-8ffb-38b8302912a2">
<img width="1114" alt="Screenshot 2023-11-30 at 12 01 35 PM" src="https://github.com/MetaMask/metamask-extension/assets/4448075/839aed19-8cf0-4a6a-a3cd-aa6062a77231">


## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
